packages/common/core/src/form/method.controller.ts

```diff
@@ -0,0 +1,125 @@
+import type { OdsCommonFieldValidityState } from './interfaces/attributes';
+import type { OdsCommonFieldMethod } from './interfaces/methods';
+import type { ODS_COMMON_INPUT_TYPE } from './constants/ods-common-input-type';
+import type { OdsFormForbiddenValues } from './validation/ods-form-forbidden-values';
+import type { OdsInputValue } from './ods-input-value';
+import { OdsGetValidityState } from './validation/ods-get-validity-state';
+
+interface OdsCommonFieldComponent {
+  defaultValue?: OdsInputValue | string | Date | HTMLInputElement['value'];
+  disabled?: boolean;  
+  emitFocus?: () => void;
+  forbiddenValues?: OdsFormForbiddenValues<string | number>;
+  hasFocus?: boolean
+  inputEl?: HTMLInputElement;
+  masked?: boolean;
+  tabindex?: number;
+  type?: ODS_COMMON_INPUT_TYPE;
+  value?: OdsInputValue | string | Date | HTMLInputElement['value'];
+}
+
+class OdsCommonFieldMethodController implements OdsCommonFieldMethod {
+
+  constructor(private readonly component: OdsCommonFieldComponent) { }
+
+  /**
+   * empty the value
+   */
+  async clear(): Promise<void> {
+    if (this.component.disabled) {
+      return;
+    }
+    this.component.value = '';
+    if (this.component.inputEl) {
+      this.component.inputEl.value = '';
+    }
+  }
+
+  /**
+   * active the focus on the input in order to let the user write something
+   */
+  async setFocus(): Promise<void> {
+    if (this.component.disabled) {
+      return;
+    }
+    this.component.inputEl?.focus();
+    this.component.hasFocus = true;
```

Comment:
seems weird to set the `hasFocus` to true if the previous line fails (ie, no `inputEl` found)